src/_pytest/main.py

```diff
@@ -206,7 +206,9 @@ def wrap_session(config, doit):
     initstate = 0
     try:
         try:
-            config._do_configure()
+            with warnings.catch_warnings():
+                warnings.simplefilter("default")
+                config._do_configure()
```

Comment:
Better to put this into `_do_configure` - just around the hook call I think.
Makes it clearer that it is for the hook only.

Reply:
Done! :)